src/lib/core/TLVDebug.cpp

```diff
@@ -353,11 +353,8 @@ CHIP_ERROR Dump(const TLVReader & aReader, DumpWriter aWriter)
 {
     void * context          = nullptr;
     DumpContext dumpContext = { aWriter, context };
-    CHIP_ERROR retval;
 
-    retval = Utilities::Iterate(aReader, DumpHandler, &dumpContext);
-
-    return retval;
+    return Utilities::Iterate(aReader, DumpHandler, &dumpContext).NoErrorIf(CHIP_END_OF_TLV);
```

Comment:
Oh, fun.  Utilities::Iterate returning an error when it completes successfully is ... something.

Followup: we should consider changing Iterate to return CHIP_NO_ERROR when there is no error, or we should add this nicety in more places.

In fact, as a second followup, most places that check for CHIP_END_OF_TLV can probably use this new helper.

Reply:
Tracked @ https://github.com/project-chip/connectedhomeip/issues/41803